model/waterfall.go

```diff
@@ -89,15 +90,40 @@ func getBuildDisplayNames(match bson.M) bson.M {
 }
 
 // This pipeline matches on versions that have a build with an ID or display name that matches variants
-func getBuildVariantFilterPipeline(variants []string, match bson.M) []bson.M {
+func getBuildVariantFilterPipeline(ctx context.Context, variants []string, match bson.M, projectId string) ([]bson.M, error) {
 	pipeline := []bson.M{}
 	match[bsonutil.GetDottedKeyName(VersionBuildVariantsKey, VersionBuildStatusDisplayNameKey)] = bson.M{"$exists": true}
 	pipeline = append(pipeline, bson.M{"$match": match})
 	matchCopy := bson.M{}
 	for key := range match {
 		matchCopy[key] = match[key]
 	}
-	pipeline = append(pipeline, getBuildDisplayNames(matchCopy))
+
+	// TODO DEVPROD-15118: Delete conditional getBuildDisplayNames check
+	mostRecentVersion, err := GetMostRecentWaterfallVersion(ctx, projectId)
+	if err != nil {
+		return []bson.M{}, errors.Wrap(err, "getting most recent version")
+	}
+
+	if mostRecentVersion.RevisionOrderNumber < MaxWaterfallVersionLimit {
+		pipeline = append(pipeline, getBuildDisplayNames(matchCopy))
```

Comment:
might be a silly question haha, but I was wondering why this condition would mean that we have to fetch the build display names?

I was thinking, if it's a new project created like a month ago and has less than 300 commits, would we need to append the display names or would it already have them?

Reply:
Ah good catch, I had added this before adding the `lastSearchableVersion == nil` check which handles the case I was aiming for here. Thanks!